tools/android_lint/pubspec.yaml

```
@@ -1,4 +1,7 @@
 name: android_lint
+environment:
+  sdk: '>=2.11.0 <3.0.0'
```

android_lint looks like it has been migrated, but it is being run with the checked-in SDK from the Dart SDK which currently is reporting version 2.12.0-0.0.dev. Hopefully that is high enough.

```suggestion
  sdk: '>=2.12.0-0.0.dev <3.0.0'
```

---

I'll make the change to the PR to see how it goes.

---

This seems to be causing issues when running `pub get` for this package it warns that package process hasn't been migrated, and then running `bin/main.dart` on the bot fails to run in sound null-safe mode because its dependencies haven't been migrated https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20Android%20Debug%20Engine/17212/overview:
```
Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:args
 - package:path
 - package:process
 - package:platform
 - package:file
 - package:meta

For solutions, see https://dart.dev/go/unsound-null-safety
```

---

perhaps tools/android_lint bin/main.dart has to be run with `--no-sound-null-safety`.

---

see #23151

---

It's easier to refactor the Dart code than to find all the places on CI that might invoke this

---

thanks @dnfield , got it!